src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/UnformattedRemappingCSharpCodeActionResolver.cs

```diff
@@ -111,25 +111,27 @@ internal class UnformattedRemappingCSharpCodeActionResolver : CSharpCodeActionRe
                 return codeAction;
             }
 
-            textEdit.Range = originalRange;
+            textEdit = textEdit with { Range = originalRange };
 
-            var codeDocumentIdentifier = new VersionedTextDocumentIdentifier()
+            var codeDocumentIdentifier = new OptionalVersionedTextDocumentIdentifier()
             {
                 Uri = csharpParams.RazorFileUri,
                 Version = documentVersion
             };
 
-            resolvedCodeAction.Edit = new WorkspaceEdit()
+            resolvedCodeAction = resolvedCodeAction with
             {
-                DocumentChanges = new[] {
+                Edit = new WorkspaceEdit()
+                {
+                    DocumentChanges = new[] {
                     new WorkspaceEditDocumentChange(
                         new TextDocumentEdit()
                         {
                             TextDocument = codeDocumentIdentifier,
                             Edits = new[] { textEdit },
```

Comment:
Wonder how you feel, stylistically, about embedding the `textEdit with { .. }` here, instead of re-assigning above?

Reply:
Stylistically it's a bit too many curly braces for me and a bit harder to debug to see the final form of text edits being included.